### ISSUE TITLE
False positive on "Return can be lifted out of match" inspection for never types

### ISSUE THREAD

**Issue by dmarcuse** [intellij-rust/intellij-rust#5418]
## Environment

* **IntelliJ Rust plugin version:** 0.2.122.3105-201
* **Rust toolchain version:** 1.43.1
* **IDE name and version:** CLion 2020.1.1
* **Operating system:** Arch Linux 5.6.12

## Problem description

When using a match statement with a never type (e.g. a binding of type `Infallible`), the IDE highlights it with a warning, suggesting that a `return` can be lifted out of the match. However, there's no `return` at all, and even when a `return` is prepended to the statement, the warning will remain.

## Steps to reproduce

The following code will reproduce the issue:

```rust
use std::convert::Infallible;

pub fn main() {
    let f = || -> Infallible { panic!() };
    match f() {}
}
```

Additionally, the suggestion will convert the above to this code, which won't actually get rid of the warning:

```rust
use std::convert::Infallible;

pub fn main() {
    let f = || -> Infallible { panic!() };
    return match f() {}
}
```

(The closure here is just to prevent the "unreachable code" warning from triggering, as it does with `let _: Infallible = panic!()`)
